src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (Bulldozer.class == type) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<T> machines = (List<T>) bulldozerProducer.get();
```

Comment:
Try to avoid repeating code.

You can initialize generic list before `if` statement: 
```java
List<? extends Machine> machines = new ArrayList<>();
```
Then add elements for each case:
```java
machines = new BulldozerProducer().get();
```
And finally cast and return result:
```java
return (List<Machine>) machines;
```